interfaces/cython/cantera/utils.pyx

```diff
@@ -117,5 +125,209 @@ cdef anyvalue_to_python(string name, CxxAnyValue& v):
+            elif itype == numbers.Integral:
+                v = list_int_to_anyvalue(item)
+            elif itype == numbers.Real:
+                v = list_double_to_anyvalue(item)
+            elif itype == dict:
+                v = list_dict_to_anyvalue(item)
+            else:
+                v = list_to_anyvalue(item)
+        elif ndim == 2:
+            if itype == str:
+                v = list2_string_to_anyvalue(item)
+            elif itype == bool:
+                v = list2_bool_to_anyvalue(item)
+            elif itype == numbers.Integral:
+                v = list2_int_to_anyvalue(item)
+            elif itype == numbers.Real:
+                v = list2_double_to_anyvalue(item)
+            else:
+                v = list_to_anyvalue(item)
+        else:
+            v = list_to_anyvalue(item)
+    elif isinstance(item, str):
+        v = stringify(item)
+    elif isinstance(item, bool):
+        v = <cbool>(item)
+    elif isinstance(item, int):
+        v = <long int>(item)
+    elif isinstance(item, float):
+        v = <double>(item)
+    elif item is None:
+        pass  # None corresponds to "empty" AnyValue
+    elif name is not None:
+        raise CanteraError("Unable to convert item of type {!r}"
+            " with key {!r} to AnyValue".format(type(item), name))
+    else:
+        raise CanteraError("Unable to convert item of type {!r}"
+            " to AnyValue".format(type(item)))
+    return v
+
+# Helper functions for converting specific types to AnyValue
+
+cdef vector[CxxAnyValue] list_to_anyvalue(data) except *:
+    cdef vector[CxxAnyValue] v
+    v.resize(len(data))
+    cdef size_t i
+    for i, item in enumerate(data):
+        v[i] = python_to_anyvalue(item)
+    return v
+
+cdef vector[double] list_double_to_anyvalue(data):
+    cdef vector[double] v
+    v.resize(len(data))
+    cdef size_t i
+    for i, item in enumerate(data):
+        v[i] = <double>item
+    return v
+
+cdef vector[long] list_int_to_anyvalue(data):
+    cdef vector[long] v
+    v.resize(len(data))
+    cdef size_t i
+    for i, item in enumerate(data):
+        v[i] = <long>item
+    return v
+
+cdef vector[cbool] list_bool_to_anyvalue(data):
+    cdef vector[cbool] v
+    v.resize(len(data))
+    cdef size_t i
+    for i, item in enumerate(data):
+        v[i] = <cbool>item
+    return v
+
+cdef vector[string] list_string_to_anyvalue(data):
+    cdef vector[string] v
+    v.resize(len(data))
+    cdef size_t i
+    for i, item in enumerate(data):
+        v[i] = stringify(item)
+    return v
+
+cdef vector[CxxAnyMap] list_dict_to_anyvalue(data) except *:
+    cdef vector[CxxAnyMap] v
+    v.resize(len(data))
+    cdef size_t i
+    for i, item in enumerate(data):
+        v[i] = dict_to_anymap(item)
+    return v
+
+cdef vector[vector[double]] list2_double_to_anyvalue(data):
+    cdef vector[vector[double]] v
+    v.resize(len(data))
+    cdef size_t i, j
+    for i, item in enumerate(data):
+        v[i].resize(len(item))
```

Comment:
My first thought here was there must be a way to optimize this, but I think this is this way because of the possibility of ragged arrays. Can you add a comment to that effect?